doc/nvim-tree-lua.txt

```diff
@@ -1152,35 +1243,52 @@ NvimTreeBookmark
 
 
 ==============================================================================
- 7. EVENTS                                                  *nvim-tree-events*
+ 8. EVENTS                                                  *nvim-tree-events*
 
 |nvim_tree_events|
 
 nvim-tree will dispatch events whenever an action is made. These events can be
 subscribed to through handler functions. This allows for even further
 customization of nvim-tree.
 
-A handler for an event is just a function which receives one argument -
-the payload of the event. The payload is different for each event type. Refer
+A handler for an event is just a function which receives one argument, the 
+payload of the event. The payload is different for each event type. Refer
 to |nvim_tree_registering_handlers| for more information.
-<
 
 |nvim_tree_registering_handlers|
 
-Handlers are registered by calling the `on_*` functions available in the
-`require('nvim-tree.events')` module. See |nvim-tree.events|.
-
+Handlers are registered by calling the `events.subscribe` function available in the
+`require("nvim-tree.api")` module.
 
-For example, registering a handler for when a node is renamed is done like this: >
-
-    local events = require('nvim-tree.events')
+For example, registering a handler for when a node is renamed is done like this:
+>
+    local api = require('nvim-tree.api')
 
-    events.on_node_renamed(function(data)
+    api.events.subscribe(Event.NodeRenamed, function(data)
         print("Node renamed from " .. data.old_name .. " to " ..  data.new_name)
     end)
+<
+
+|nvim_tree_events_kind|
+
+You can access the event enum with:
+>
+    local Event = require('nvim-tree.api').events.Event
+<
+Here is the list of available variant of this enum:
```

Comment:
We need the payload / params documented. You could just copy the descriptions from "7.1 Lua module: nvim-tree.events".